asapdiscovery-genetics/asapdiscovery/genetics/blast.py

```diff
@@ -0,0 +1,264 @@
+                        if verbose:
+                            print(
+                                f"length {hsps0.identities}, score {pidentity}: {align.title}"
+                            )
+                        data = {
+                            "query": [query],
+                            "ID": [id],
+                            "description": [description],
+                            "sequence": [sequence_to_model],
+                            "score": [pidentity],
+                        }
+                        df_row = pd.DataFrame(data)
+                        dfs.append(df_row)
+                df = pd.concat(dfs, axis=0, ignore_index=True, sort=False).dropna(
+                    axis=1, how="all"
+                )
+    return df
+
+
+def get_blast_seqs(
+    seq_source: str,
+    save_folder: str,
+    input_type="fasta",
+    nhits=100,
+    nalign=500,
+    database="refseq_protein",
+    xml_file="results.xml",
+    verbose=True,
+    save_csv=None,
+) -> pd.DataFrame:
+    """Run a BLAST search on a protein sequence.
+    Args:
+        seq_source (str): Source with the sequence.
+        input_type (str, optional): Type of sequence source ["pre-cal", "fasta", "sequence"]. Defaults to "fasta".
+        save_file (str, optional): Name of output file storing BLAST results. Defaults to "results.xml".
+
+    Args:
+        seq_source (str): Source with the sequence.
+        save_folder (str): Path to folder to save BLAST results
+        input_type (str, optional): Type of sequence source ["pre-cal", "fasta", "sequence"]. Defaults to "fasta".
+        nhits (int, optional): Number of hits, hitlist_size parameter in BLAST. Defaults to 100.
+        nalign (int, optional): Number of alignments, alignments parameter in BLAST. Defaults to 500.
+        database (str, optional): Name of BLAST database. Defaults to "refseq_protein".
+        xml_file (str, optional): Name to be given to XML with BLAST results. Defaults to "results.xml".
+        verbose (bool, optional): Whether to print info on BLAST search. Defaults to True.
+        save_csv (Union[str, None], optional): CSV file name to optionally save dataframe. Defaults to None.
+
+    Returns:
+        pd.DataFrame: DataFrame with Blast results.
+    """
+
+    from pathlib import Path
+
+    if input_type == "pre-calc":
+        matches_df = parse_blast(seq_source, verbose)
+        return matches_df
+    elif input_type == "fasta":
+        # Input is file name
+        sequence = open(seq_source).read()
+    elif input_type == "sequence":
+        # Input is sequence
+        sequence = seq_source
+    else:  # Another source?
+        return
+
+    # Retrieve blastp results
+    program = "blastp"  # protein sequence BLAST
+    database = database  # protein database
+    alignments = nalign  # number of alignments to retrieve
+    result_handle = NCBIWWW.qblast(
+        program, database, sequence, hitlist_size=nhits, alignments=alignments
+    )
```

Comment:
Just pass these in directly 

```suggestion
    result_handle = NCBIWWW.qblast(
        "qblast", database, sequence, hitlist_size=nhits, alignments=nalign
    )
```